pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -112,7 +111,12 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1.ApiServerSour
 	}
 	source.Status.PropagateDeploymentAvailability(ra)
 
-	source.Status.CloudEventAttributes = r.createCloudEventAttributes()
+	cloudEventAttributes, err := r.createCloudEventAttributes(source)
```

Comment:
Do we need to handle empty case for backward compatibility? Or should we add validation or webhook to ensure this field is always valid?

Reply:
We already have validation in webhook to ensure this field is always valid.

Reply:
https://github.com/knative/eventing/blob/f0dc15b0d493dba893d7a14c2292e335b1d77bb9/pkg/apis/sources/v1/apiserver_validation.go#L43

Reply:
Make sense.